contracts/mocks/ERC1155ReceiverMock.sol

```diff
@@ -7,9 +7,13 @@ import "../utils/introspection/ERC165.sol";
 
 contract ERC1155ReceiverMock is ERC165, IERC1155Receiver {
     bytes4 private _recRetval;
-    bool private _recReverts;
+    uint private constant _RECREVERTS_FALSE = 1;
+    uint private constant _RECREVERTS_TRUE = 2;
+    uint private _recReverts;
     bytes4 private _batRetval;
-    bool private _batReverts;
+    uint private constant _BATREVERTS_FALSE = 1;
+    uint private constant _BATREVERTS_TRUE = 2;
+    uint private _batReverts;
```

Comment:
After that the number of slots required is 4, since: 
- 0: `bytes4 private _recRetval;`
- 1: `uint private _recReverts;`
- 2: `bytes4 private _batRetval;`
- 3: `uint private _batReverts;`

But in the initial version we're using only a single storage slot. 

Changing `bool` to `uint`  does make sense, since stack operations with bools require casting 32 bytes slots and that occurs overhead, but it's not a decent number in comparison.

Reply:
Changing the value of a bool variable from false to true costs 20,000 units of gas, which is much more than changing the value of a non-zero integer variable.

Reply:
changing the value of a bool variable from false to true doesn't cost 20,000 units of gas, since the same slot contains another variables (packed variables with the current bool) i.e. the slot != 0. 

Pay attention on the fact, that in the same slot we have 4 variables and each of them have their own value.

Reply:
I reverted these changes, only leaving cases where a bool field occupies a whole slot.